index.js

```diff
@@ -251,7 +280,8 @@ URL.prototype.toString = function toString(stringify) {
     result += '@';
   }
 
-  result += url.hostname;
+  // rough check of IPv6 Hostname (v4 doesn't include colon)
+  result += ~url.hostname.indexOf(':') ? '[' + url.hostname + ']' : url.hostname;
```

Comment:
We can probably replace this line and the next one with just `result += url.host`;